### ISSUE TITLE
Matching broken by changes in Nim devel branch

### ISSUE THREAD

**Issue by nigredo-tori** [andreaferretti/patty#6]
Recent changes in Nim's devel branch break patty's macros. For example, the following code now doesn't compile:
```nim
import patty

variantp Foo:
  Bar

var f = Bar()

match f:
  Bar:
    echo "this is Bar"
```
result:
```
stack trace: (most recent call last)
/home/dmitry/.nimble/pkgs/patty-0.1.7/patty.nim(364) match
/home/dmitry/.nimble/pkgs/patty-0.1.7/patty.nim(364) match(191) isObject
test_patty.nim(8, 7) template/generic instantiation from here
/home/dmitry/.nimble/pkgs/patty-0.1.7/patty.nim(364) match(191) isObject(191, 38) Error: index out of bounds
```

bisect points to this commit: https://github.com/nim-lang/Nim/commit/0356f53b5619782c650f56a6842cbaf91a82137d

**Comment by andreaferretti** [comment]
This is very unfortunate. I will have a look into this

**Comment by andreaferretti** [comment]
Fixed with Patty 0.1.8
